visualization/app/codeCharta/state/fileState.service.ts

```diff
@@ -27,7 +29,12 @@ export class FileStateService {
 	}
 
 	public getFileStates(): FileState[] {
-		return this.fileStates
+		return _.cloneDeep(this.fileStates)
```

Comment:
remove deepClone. DeepCloning can cost some time and we should only mutate the fileStates with the methods provided by this service (or later with the actions defined for this part of the store)

Reply:
I just tested it with one of the largest maps we have, and i could not tell a performance difference compared to the current master. There also only remain 3 cases in which the FileState needs to be obtained; one of which needs a deep copy anwyays (see bug above). Personally i kind of like that with the deep clone we cannot alter file states by accident. If you dislike it conceptually, we can remove it from this method and deep clone on the recieving end where necessary.

Reply:
...well changed my mind; in the event we get it by reference anyways, and there is no point in preventing pass by reference when getting it while we pass it by reference with the action. I will remove deep clone and add it in the preRenderService, where we actually need it.